tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/AbstractPortalBundle.java

```diff
@@ -0,0 +1,271 @@
+package com.liferay.ide.server.core.portal;
+
+import com.liferay.ide.core.ILiferayConstants;
+import com.liferay.ide.core.util.CoreUtil;
+import com.liferay.ide.core.util.StringPool;
+import com.liferay.ide.server.core.LiferayServerCore;
+import com.liferay.ide.server.util.LiferayPortalValueLoader;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Properties;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import org.eclipse.core.runtime.IPath;
+import org.osgi.framework.Version;
+
+/**
+ * @author Simon Jiang
+ */
+public abstract class AbstractPortalBundle implements PortalBundle
+{
+    
+    private static final String CONFIG_TYPE_SERVER = "server"; //$NON-NLS-1$
+    private static final String CONFIG_TYPE_VERSION = "version"; //$NON-NLS-1$
+    private static final Version MANIFEST_VERSION_REQUIRED = ILiferayConstants.V620;
+
+
+    protected IPath autoDeployPath;
+    protected IPath liferayHome;
+    protected IPath modulesPath;
+    protected IPath bundlePath;
+    protected String version;
+    protected int jmxRemotePort;
+ 
+    public AbstractPortalBundle( IPath path )
+    {
+        if( path == null )
+        {
+            throw new IllegalArgumentException( "path cannot be null" );
+        }
+
+        this.bundlePath = path;
+        
+        this.liferayHome = bundlePath.append( ".." );
```

Comment:
Will this be true for all bundles?  I guess we can look at glassfish and jetty bundles.